### ISSUE TITLE
Android 12 Support

### ISSUE THREAD

**Issue by amedfa** [frap129/android_packages_apps_Eink_SystemUI#1]
This post https://www.tomesoftware.com/labs/customizing-androids-systemui/ starts with

>Last month I wrote a bit about my work on porting Android 12 to the PineNote, an open source E-Ink tablet.

I been able to compile your project on an Android11 source code, but the structure of Android12 regard to the daggers  looks different

do you have any local update for running this project on Android 12.

Thanks in advance

**Comment by frap129** [comment]
Nope. I was able to get it mostly compiling, but ultimately it failed due to kernel version requirements. The kernel provided by rockchip was already EOL on Android 11, so moving to android 12 would've been a lot of porting work. I haven't checked on the PineNote status in a while, maybe the mainline kernel has enough hw support that it's worth looking into.

If you're working on a different device and just need this repo to work on Android 12, porting it shouldn't be too hard. A lot of changes will need to be made to the code that lets you hook into SystemUI-Core,  but I don't think any of the ui itself will need changing
